File: NoteTranscription/Core Model/Note+Convenience.m
//
//  Note+Convenience.m
//  NoteTranscription
//
//  Created by Josh Kocsis on 10/25/20.
//

#import "Note+Convenience.h"
#import <CoreData/CoreData.h>
#import "NoteTranscription-Swift.h"

@class BW4CoreDataStack;

@implementation Note (Convenience)

- (instancetype)initWithTitle:(NSString *)title bodyText:(NSString *)bodyText timestamp:(NSDate *)timestamp img:(NSData *)img 
{
    self = [self initWithContext:BW4CoreDataStack.sharedStack.mainContext];
    if (self) {
        self.title = [title copy];
        self.bodyText = [bodyText copy];
        self.timestamp = [timestamp copy];
        self.img = [img copy];
    }
    return self;
}

@end

File: NoteTranscription/Core Model/Note+Convenience.swift
//
//  Note+Convenience.swift
//  NoteTranscription
//
//  Created by Josh Kocsis on 10/19/20.
//

import Foundation
import CoreData


extension Note {
    @discardableResult convenience init(title: String?,
                                    bodyText: String?,
                                    timestamp: Date? = Date(),
                                    context: NSManagedObjectContext = CoreDataStack.shared.mainContext) {
    self.init(context: context)
    self.title = title
    self.bodyText = bodyText
    self.timestamp = timestamp
    }
}
